### DISCUSSION TITLE
Breakpoints question

### DISCUSSION THREAD

**Discussion by dixonge** [tailwindlabs/tailwindcss#1459] [General]
So, evidently, if I set a breakpoint for large (like lg:w-48) but do *not* set a breakpoint for mobile as a default (w-16) then the large breakpoint will take effect even for mobile view. Do I have that right? That is what I just figured out on my own, but didn't seem to find an explanation for in the docs.

Is this a result of not including the ```@import tailwindcss/base``` in my css ?

**Comment by adamwathan** [comment]
Nope that is not the case, see this demo:

https://jsfiddle.net/zLnyet31/

Can you share a reproduction of what you're seeing? I'm sure we can figure out what's happening 👍

**Comment by benface** [comment]
> So, evidently, if I set a breakpoint for large (like lg:w-48) but do not set a breakpoint for mobile as a default (w-16) then the large breakpoint will take effect even for mobile view. Do I have that right? 

No. If you have `lg:w-48`, `w-48` will be applied on breakpoint `lg` and up.

> Is this a result of not including the @import tailwindcss/base in my css ?

No, Preflight has nothing to do with variants / responsive utilities. The `lg:w-48` class rule is defined like this in the CSS:

```css
@media (min-width: 1024px) {
  .lg\:w-48 {
    width: 12rem;
  }
}
```

That rule is generated where `@tailwind screens` appears, or if you don't have that, it will be at the end of `@tailwind utilities`.

EDIT: Ninja'd by Adam. xD

**Comment by dixonge** [comment]
I was using lg:w-9/12 - this shows up in index.css in the appropriate @media section.

If used by itself, according to dev tools, even at 320px of screen width, the element width was 75%. Only when I also added w-full did the mobile view stretch out to a full 100% width. 

I'm using this with eleventy, postcss, purgecss, etc. Hosting is Netlify. 

I already fixed things by adding the default widths, but I set it back on my local dev long enough to grab a screenshot for you. As you can see, the only breakpoint setting that is struckthrough is one that has been over-ridden by specifying a default setting...

![tailwind](https://user-images.githubusercontent.com/1578506/77007448-db2daa00-6929-11ea-956b-03495a157344.jpg)

**Comment by adamwathan** [comment]
Maybe missing the necessary viewport meta tags to get the browser to interpret the dimensions correctly?

```html
 <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
```

**Comment by dixonge** [comment]
@adamwathan I had that but was missing the 'shrink-to-fit' part - so I added it, then removed one of the 'w-full' classes. All of the 'lg:' classes were still taking effect on mobile on that element.

The CSS that gets processed by postcss starts with:

```
@import "tailwindcss/components";

@import "tailwindcss/utilities";
```

Is this not right?

**Comment by adamwathan** [comment]
Can you share a project that reproduces? Much easier to debug.

**Comment by florianbouvot** [comment]
Why some classes refer to `index.css:...` and others from `(index):36` ?

**Comment by dixonge** [comment]
[the project in question](https://github.com/dixonge/vagabondians-11ty)

**Comment by dixonge** [comment]
@florianbouvot I noticed that too, and I have no idea why

**Comment by adamwathan** [comment]
The problem is you have all these hard-coded styles in your head:

https://github.com/dixonge/vagabondians-11ty/blob/master/_includes/layouts/head.njk#L13

These styles contain a bunch of copies of the Tailwind breakpoint-specific utilities but with the media queries removed, causing the styles to be applied all the time.

This happened in this commit:

https://github.com/dixonge/vagabondians-11ty/commit/9c2182ff4309d8980fff4849665ccd2505acf434

...where you pasted those styles in yourself with no media queries, not sure where they came from.

**Comment by dixonge** [comment]
well crap! I did that as part of running through reported issues from Lighthouse testing - was trying to split out preload vs load later. Never occurred to me how badly this would work with Tailwind and purgecss. *facepalm*

**Comment by masterwww1** [comment]
Good sample
